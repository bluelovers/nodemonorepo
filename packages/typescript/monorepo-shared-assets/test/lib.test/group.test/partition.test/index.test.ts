import partition from '../../../../lib/group/partition'

it('using list of classifiers', () => {
  type Classifier = partition.Classifier<any>

  const list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
    .map(x => [x, String(x), Symbol(x), [x], {x}, Boolean(x & 1)])
    .reduce((prev, current) => [...prev, ...current])

  const classifiers: Classifier[] = [
    ['number', x => typeof x === 'number'],
    ['string', x => typeof x === 'string'],
    ['symbol', x => typeof x === 'symbol'],
    ['array', x => x instanceof Array],
    ['object', x => x && typeof x === 'object']
  ]

  const result = partition.fromClassifierList(list, classifiers)
  expect(result).toMatchSnapshot()
  expect(result).toEqual(partition(list, classifiers)) // default method
})

it('using dict of classifiers', () => {
  type Dict = partition.Classifier.Dict<any>

  const list = [
    0, 1, 2, 3,
    ...'abcd',
    4, 5, 6, 7,
    ...'ghijkl',
    8, 9
  ]

  const classifiers: Dict = {
    oddNumbers: x => isFinite(x) && Boolean(x & 1),
    evenNumbers: x => isFinite(x) && !Boolean(x & 1)
  }

  expect(partition.fromClassifierDict(list, classifiers)).toMatchSnapshot()
})
