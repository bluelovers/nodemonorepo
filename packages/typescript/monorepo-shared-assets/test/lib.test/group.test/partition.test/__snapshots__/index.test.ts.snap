// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using dict of classifiers 1`] = `
Object {
  "classified": Object {
    "evenNumbers": Set {
      0,
      2,
      4,
      6,
      8,
    },
    "oddNumbers": Set {
      1,
      3,
      5,
      7,
      9,
    },
  },
  "untouched": Set {
    "a",
    "b",
    "c",
    "d",
    "g",
    "h",
    "i",
    "j",
    "k",
    "l",
  },
}
`;

exports[`using list of classifiers 1`] = `
Object {
  "classified": Object {
    "array": Set {
      Array [
        0,
      ],
      Array [
        1,
      ],
      Array [
        2,
      ],
      Array [
        3,
      ],
      Array [
        4,
      ],
      Array [
        5,
      ],
      Array [
        6,
      ],
      Array [
        7,
      ],
      Array [
        8,
      ],
      Array [
        9,
      ],
    },
    "number": Set {
      0,
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
    },
    "object": Set {
      Object {
        "x": 0,
      },
      Object {
        "x": 1,
      },
      Object {
        "x": 2,
      },
      Object {
        "x": 3,
      },
      Object {
        "x": 4,
      },
      Object {
        "x": 5,
      },
      Object {
        "x": 6,
      },
      Object {
        "x": 7,
      },
      Object {
        "x": 8,
      },
      Object {
        "x": 9,
      },
    },
    "string": Set {
      "0",
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
    },
    "symbol": Set {
      Symbol(0),
      Symbol(1),
      Symbol(2),
      Symbol(3),
      Symbol(4),
      Symbol(5),
      Symbol(6),
      Symbol(7),
      Symbol(8),
      Symbol(9),
    },
  },
  "untouched": Set {
    false,
    true,
  },
}
`;
